app/src/main/res/layout/activity_main.xml

```diff
@@ -1,65 +1,54 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".MainActivity">
-
-    <Button
-        android:id="@+id/button_toast"
-        android:layout_width="0dp"
+    
+    <TextView
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:layout_marginStart="8dp"
-        android:layout_marginTop="8dp"
-        android:layout_marginEnd="8dp"
+        android:id="@+id/article_heading"
```

Comment:
1. why do we use @+id instead of @id
2. why do we use @string instead of @+string or @dimen instead of @+dimen?

Reply:
1. @id doesn't create an id if it doesn't exist
2. If a string/dimen doesn't exist, error will be displayed.